tests/foreman/destructive/test_fm_upgrade.py

```diff
@@ -0,0 +1,51 @@
+"""Destructive test module for satellite-maintain upgrade functionality
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseComponent: SatelliteMaintain
+
+:Team: Platform
+
+:CaseImportance: Critical
+
+"""
+
+import pytest
+
+pytestmark = pytest.mark.destructive
+
+
+@pytest.mark.include_capsule
+def test_negative_ipv6_update_check(sat_maintain):
+    """Ensure update check fails when ipv6.disable=1 in boot options
+
+    :id: 7b3e017f-443a-4204-99be-e39fa04c89f6
+
+    :parametrized: yes
+
+    :steps:
+        1. Add ipv6.disable to grub boot options
+        2. Reboot
+        3. Run update check
+
+    :expectedresults: Update check fails due to ipv6.disable=1 in boot options
+    """
+    result = sat_maintain.execute('grubby --args="ipv6.disable=1" --update-kernel=ALL')
```

Comment:
Does `sat_maintain` here represent the CLI tool or a host? As my question hopefully implies, I find this very confusing :)

Reply:
Yeah it's not my favorite design but it makes testing for capsule vs satellite very easy since satellite-maintain has the same commands for both.

https://github.com/SatelliteQE/robottelo/blob/00ce6d0672a1384511dcfa563c251fe988f12d5f/pytest_fixtures/component/maintain.py#L25

The `sat_maintain` fixture spins up a Satellite and Capsule, and then the correct one is returned based on which test you're running. You'll see in this scenario I have `include_capsule` on https://github.com/SatelliteQE/robottelo/pull/15062/files#diff-f514a11bb5043f68643019fff6bf6af426fb404a3525bc49ec3b4af920234373R20. This parameterizes this test to run once on a Satellite and once on a Capsule. The `sat_maintain` fixture already made one of each so we just pass along the right one for each test.

Reply:
this can be improved with better naming :) (something I'm bad at)